src/env.rs

```diff
@@ -60,22 +60,18 @@ impl OwnedEnv {
         closure(env)
     }
 
-    /// Run a closure that produces a term, then send the term to another process.
+    /// Send a message to another process.
     ///
-    /// After the closure runs and the message is sent, the environment is cleared as though by
-    /// calling the `.clear()` method.
-    pub fn send<F>(&mut self, recipient: ErlNifPid, closure: F)
-        where F: for<'a> FnOnce(NifEnv<'a>) -> NifTerm<'a>
-    {
-        let env_lifetime = ();
+    /// The environment is cleared as though by calling the `.clear()` method.
+    ///
+    /// # Unsafe
+    ///
+    /// This must not be called from a thread that is managed by the Erlang VM.
+    /// Also, `message` must be a valid term that belongs to this environment.
+    pub unsafe fn send(&mut self, recipient: &NifPid, message: NIF_TERM) {
```

Comment:
Would it make sense to use `enif_thread_type` here instead of having it be unsafe? I don't believe that function is very expensive.